tests/phpunit/tests/formatting/redirect.php

```diff
@@ -43,6 +43,96 @@ public function data_wp_redirect_bad_status_code() {
 		);
 	}
 
+	/**
+	 * @ticket 60208
+	 *
+	 * @dataProvider data_wp_redirect_loop
+	 *
+	 * @covers ::wp_redirect
+	 * @covers ::wp_safe_redirect
+	 * @covers ::_wp_is_self_redirect
+	 *
+	 * @param string $location The path or URL to redirect to.
+	 * @param bool   $ssl like HTTPS
+	 * @param string $host like HTTP_HOST
+	 * @param string $request_uri like REQUEST_URI
+	 * @param bool   $expected true if it should redirect
+	 * @param bool   $safe_expected optional expected value of safe_redirect
+	 */
+	public function test_wp_redirect_loop( $location, $ssl, $host, $request_uri, $expected, $safe_expected = null ) {
```

Comment:
Rather than doing a lot of heavy lifting here, would e2e tests in playwright make more sense?